database/seeds/ModelSeeders/UserSeeder.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+use Illuminate\Database\Seeder;
+use Faker\Factory as Faker;
+
+class UserSeeder extends Seeder
+{
+    /**
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+        // DB::table('phpbb_users')->delete();
+        // DB::table('osu_user_stats')->delete();
+        // DB::table('osu_user_stats_fruits')->delete();
+        // DB::table('osu_user_stats_mania')->delete();
+        // DB::table('osu_user_stats_taiko')->delete();
+        // DB::table('osu_user_performance_rank')->delete();
+
+        $this->beatmapCount = App\Models\Beatmap::count();
+        $this->faker = Faker::create();
+        if ($this->beatmapCount === 0) {
+            $this->command->info('Can\'t seed events, scores, events or favourite maps due to having no beatmap data.');
+        }
+
+        // Store some modifiers
+        $this->improvement_speeds = [
+            (rand(100, 110) / 100), // Fast Learner
+            (rand(100, 102) / 100), // Slow Learner
+            (rand(100, 115) / 100), // Genius / Multiaccounter :P
+        ];
+
+        // Create 10 users and their stats
+        factory(App\Models\User::class, 10)->create()->each(function ($u) {
+
+            // USER STATS
+            $common_countries = ['US', 'JP', 'CN', 'DE', 'TW', 'RU', 'KR', 'PL', 'CA', 'FR', 'BR', 'GB', 'AU'];
+            $country_code = $common_countries[array_rand($common_countries)];
+
+            $rank1 = rand(1, 500000);
+            $rank2 = rand(1, 500000);
+            $rank3 = rand(1, 500000);
+            $rank4 = rand(1, 500000);
+            $st = $u->statisticsOsu()->save(factory(App\Models\UserStatistics\Osu::class)->create(['country_acronym' => $country_code, 'rank' => $rank1, 'rank_score_index' => $rank1]));
+            $st1 = $u->statisticsOsu()->save(factory(App\Models\UserStatistics\Taiko::class)->create(['country_acronym' => $country_code, 'rank' => $rank2, 'rank_score_index' => $rank2]));
```

Comment:
this doesn't seem right to attach Taiko stats through Osu relation but actually works.

Though probably the whole thing can be looped like this?

```php
foreach (array_keys(Beatmap::modes()) as $modeStr) {
    $modeStr = studly_case($modeStr);

    $rank = rand(...)
    $st = $u->statistics($modeStr)->save(factory("...\\{$modeStr}")->...

    ...create rank here...
}
```